library/network_connections.py

```diff
@@ -972,6 +976,10 @@ def connection_create(self, connections, idx, connection_current=None):
                 s_ip6.set_property(NM.SETTING_IP_CONFIG_METHOD, "manual")
             else:
                 s_ip6.set_property(NM.SETTING_IP_CONFIG_METHOD, "ignore")
+            
+            if ip["ipv6_disabled"]:
```

Comment:
I am not convinced that introducing `ipv6_disabled` is the best approach here. As this code shows, the decision tree becomes unclear. If it is set, the method will be first "ignore" and then updated to "disabled". This is confusing.

Reply:
updated

Reply:
> As this code shows, the decision tree becomes unclear. If it is set, the method will be first "ignore" and then updated to "disabled". This is confusing.

The decision tree becomes clear if `ArgValidator` resolves all inconsistencies. Yes, `"ipv6_disabled:True"` and `"auto6:True"` conflicts, but `ArgValidator` should reject such configuration as valid and guarantee that the result of the validation is consistent.

Reply:
> I am not convinced that introducing ipv6_disabled is the best approach here.

Maybe. This question is about naming of parameters (and which form a parameter should have). The choice to have here a new boolean `ipv6_disabled` seems sensible to me. But no doubt, it could also be different.

Is this question stil unresolved?

Reply:
The problem I see is with the redundancy, for example auto6:yes implies ipv6_disabled:no. Therefore it would be nicer to find a solution where the configuration format ensures that there are no such conflicts but maybe it is not possible. I guess ipv6_link_local:yes|no has similar problems - but maybe it would be more descriptive about the impact of the option. Or are static IPs without link-local IPs a valid configuration for IPv6?

Reply:
they are not. You always need a link-local IPv6 address (except with `ipv6_disabled=yes`)

In the current branch,
```
    ipv6_disabled = no
    auto6 = no
    addresses = []
```
translates to `ipv6.method=ignored`, but it really should be instead `ipv6.method=link-local`.

I don't see a problem with "redundancy". Sure, in settings like

```
   ipv6_disabled = no
   auto6 = yes
```
the `ipv6_disabled` is redundant. But you can already do
```
   ipv6_disabled = no
   auto6 = yes
   addresses = []
```
where both `addresses` and `ipv6_disabled` is redundant too. That seems not avoidable. It would be annoying to fail:
```
    auto6 = yes
    addresses = []
```
with: "ERROR: cannot configure an empty list of addresses because that redundant to not specifying the option."


On the other hand, conflicting settings are rejected as invalid by the branch.